sls-versions/src/main/java/com/palantir/sls/versions/SlsVersion.java

```diff
@@ -43,7 +46,7 @@ public static boolean check(String coordinate) {
         try {
             valueOf(coordinate);
             return true;
-        } catch (IllegalArgumentException e) {
+        } catch (SafeIllegalArgumentException e) {
```

Comment:
I think I'd leave this as IAE just to be on the safe - they might come from inside immutables too?

Reply:
everything here depends on logsafe, and I ran into this error *because* deep down something was throwing a SIAE

It also passes all of the unit tests - if you can point to a codepath where it wouldn't work, I'm more than happy to adjust, but I think this fixes everything

Reply:
Fair enough - immutables would throw NullPointerExceptions anwyay, so this doesn't really buy us anything.